libraries/botframework-connector/src/auth/connectorFactoryImpl.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { getDefaultUserAgentValue, userAgentPolicy } from '@azure/ms-rest-js';
+import { ConnectorClient } from '../connectorApi/connectorClient';
+import { ConnectorClientOptions } from '../connectorApi/models';
+import { ConnectorFactory } from '../connectorFactory';
+import { ServiceClientCredentialsFactory } from './serviceClientCredentialsFactory';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const packageInfo: Record<'name' | 'version', string> = require('../../package.json');
+export const USER_AGENT = `${packageInfo.name}/${packageInfo.version} ${getDefaultUserAgentValue()} `;
+
+// Internal
+export class ConnectorFactoryImpl extends ConnectorFactory {
+    constructor(
+        private readonly appId: string,
+        private readonly toChannelFromBotOAuthScope: string,
+        private readonly loginEndpoint: string,
+        private readonly validateAuthority: boolean,
+        private readonly credentialFactory: ServiceClientCredentialsFactory,
+        private readonly connectorClientOptions: ConnectorClientOptions = {}
+    ) {
+        super();
+    }
+
+    async create(serviceUrl: string, audience: string): Promise<ConnectorClient> {
+        // Use the credentials factory to create credentails specific to this particular cloud environment.
+        const credentials = await this.credentialFactory.createCredentials(
+            this.appId,
+            audience ?? this.toChannelFromBotOAuthScope,
+            this.loginEndpoint,
+            this.validateAuthority
+        );
+
+        // A new connector client for making calls against this serviceUrl using credentials derived from the current appId and the specified audience.
+        const options = this.getClientOptions(serviceUrl);
+        return new ConnectorClient(credentials, options);
+    }
+
+    private getClientOptions(serviceUrl: string): ConnectorClientOptions {
```

Comment:
Pedantic copy of: https://github.com/microsoft/botbuilder-js/blob/ee4705c72d49a1e421d8e3c81dddfaaf8b548401/libraries/botbuilder/src/botFrameworkAdapter.ts#L1485

I skipped the `httpClient?: any` parameter as it was only used in the BotFrameworkAdapter to pass in a `StreamingHttpClient`:

https://github.com/microsoft/botbuilder-js/blob/ee4705c72d49a1e421d8e3c81dddfaaf8b548401/libraries/botbuilder/src/botFrameworkAdapter.ts#L1477